### PR TITLE
Feat: add ClientIdentityPenetration flag

### PR DESCRIPTION
**Pull request by Somefive**

### CODE REVIEW

File: pkg/featuregates/featue_gate.go [hunk 1]
```diff
@@ -31,9 +31,17 @@ const (
...
+	// apiserver in ManagedCluster
+	AlwaysImpersonate featuregate.Feature = "AlwaysImpersonate"
 )
 
 var DefaultKubeFedFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
-	HealthinessCheck: {Default: false, PreRelease: featuregate.Alpha},
-	SecretCache:      {Default: true, PreRelease: featuregate.Beta},
+	HealthinessCheck:  {Default: false, PreRelease: featuregate.Alpha},
+	SecretCache:       {Default: true, PreRelease: featuregate.Beta},
+	AlwaysImpersonate: {Default: false, PreRelease: featuregate.Beta},
```
[the remark this answers is not in the document]
Comment (Somefive) [reply pkg/featuregates/featue_gate.go]: Done.

File: pkg/featuregates/featue_gate.go [hunk 2]
```diff
@@ -31,9 +31,17 @@ const (
 	// SecretCache runs a namespaced secret informer inside the apiserver which
 	// provides a cache for reading secret data.
 	SecretCache featuregate.Feature = "SecretCache"
+
+	// owner: @somefive
+	// alpha: v1.4.0
+	//
+	// AlwaysImpersonate enforce impersonate as the request user when accessing
+	// apiserver in ManagedCluster
+	AlwaysImpersonate featuregate.Feature = "AlwaysImpersonate"
```
Comment (yue9944882) [inline pkg/featuregates/featue_gate.go:40]: s/AlwaysImpersonate/ClientIdentityPenestration/

Comment (Somefive) [reply pkg/featuregates/featue_gate.go]: ClientIdentityPenestration -> ClientIdentityPenetration, Done

**Review by yue9944882 — APPROVED** [review]
/lgtm

**Pull request merged**